front-end/src/store/mainPageSlice.js

```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+export const mainPageSlice = createSlice({
+  name: 'mainPage',
+  initialState: {
+    showTranscriptUploadModal: false,
+  },
+  reducers: {
+    toggleTranscriptUploadModal: state => {
+      state.showTranscriptUploadModal = !state.showTranscriptUploadModal;
+      console.log("Set showTranscriptUploadModal to " + state.showTranscriptUploadModal);
+    },
+  }
+})
+
+export const { toggleTranscriptUploadModal } = mainPageSlice.actions;
+
+export const showTranscriptUploadModal = (state) => state.showTranscriptUploadModal;
+
+export default mainPageSlice.reducer;
```

Comment:
I dont think this file is necessary

Reply:
From what I read online, each (group of) features should have its own slice so that the reducers are more organized. Maybe that's not as important for our project but I felt that it was intuitive for each page to have its own slice since user actions will be different between them.

Unless you mean the entire file is unnecessary? I'm not sure but I think without using store, moving between pages will reset any changes that were made to them by the user.

Reply:
It is definitely true that when we use Redux, we are going to want individual files for different modules of our app. However, the reason this file is not necessary is because we should be doing all this in the local state. I linked an article above with an explanation, but basically, showing the modal is considered a small-scale action, and thus we do not want that info in global state (Redux), but instead want to use local state React.useState().